python/llm/src/bigdl/llm/transformers/models/chatglm2.py

```diff
@@ -265,13 +271,24 @@ def chatglm2_attention_forward_8eb45c(
 
         if cache_k.stride()[1] <= cache_k.size(2) * cache_k.size(3):
             max_cache_length = past_length + cur_length + KV_CACHE_ALLOC_BLOCK_LENGTH
-            new_cache_k, new_cache_v = extend_kv_cache(batch_size,
-                                                       self.num_attention_heads_per_partition,
-                                                       self.hidden_size_per_attention_head,
-                                                       past_length,
-                                                       max_cache_length,
-                                                       dtype=query_layer.dtype,
-                                                       device=device)
+            if device.type == "xpu" and batch_size > 1:
+                # Optimize memory usage and support beam_search on gpu.
+                # If you use beam_search for generation, the batch_size will be larger than 1.
```

Comment:
I think you need to explain how `batch_size > 1` is handled differently from `batch_size = 1`